### ISSUE TITLE
Receiving old protobuf messages after repeated machine preparation

### ISSUE THREAD

**Issue by TarikViehmann** [robocup-logistics/rcll-refbox#88]
I encountered the following issue while working on an implementation for the storage station:
I started the refbox, entered the production phase (no robots connected) and used the `rcll-prepare-machine` tool to send machine instructions manually in order to debug my code.
My code seemed to work fine (you can find it here: https://github.com/robocup-logistics/rcll-refbox/tree/tviehmann/storage-station), but from time to time a machine would break after issuing consecutive storing operations.
I wanted to know whether it was due to the tool sending wrong stuff or it was caused by the receiving end of the protobuf communication so i added a field to the prepare message that contains the timestamp of the sender. It appears that upon sending the second message, one of the old messages arrives in clips and is processed (note that this never happens when i do a single prepare).

Here is what the `rcll-prepare-machine` command does:
```
[tarikwork@localhost bin]$ ./rcll-prepare-machine Carologistics C-SS STORE 4 0
Waiting for beacon from refbox...
Announcing machine type
prep message built with 1, 4 0
GameState received:  00:00:13.279  PRODUCTION RUNNING  0:0 points, Carologistics vs. 
Set crypto key to randomkey (cipher aes-128-cbc)
Announcing machine type
prep message built with 1, 4 0
MachineInfo received:
  C-RS1, state: IDLE
  C-DS, state: IDLE
  C-BS, state: IDLE
  C-CS2, state: IDLE
  C-CS1, state: IDLE
  C-RS2, state: IDLE
  C-SS, state: PROCESSING
[tarikwork@localhost bin]$ ./rcll-prepare-machine Carologistics C-SS STORE 4 1
Waiting for beacon from refbox...
Announcing machine type
prep message built with 1, 4 1
GameState received:  00:00:45.920  PRODUCTION RUNNING  0:0 points, Carologistics vs. 
Set crypto key to randomkey (cipher aes-128-cbc)
Announcing machine type
prep message built with 1, 4 1
GameState received:  00:00:49.920  PRODUCTION RUNNING  0:0 points, Carologistics vs. 
MachineInfo received:
  C-RS1, state: IDLE
  C-DS, state: IDLE
  C-BS, state: IDLE
  C-CS2, state: IDLE
  C-CS1, state: IDLE
  C-RS2, state: IDLE
  C-SS, state: BROKEN

```

Here is the refbox log (the timestamp of the message shown converts to 4:45:02.500 PM so it refers to the first attempt of preparation)
 ```
I 16:45:01.539741 C: Received prepare for C-SS
I 16:45:01.540323 C: C-SS is IDLE, processing prepare                           
I 16:45:01.541330 C: Machine C-SS prepared for STORE                            
I 16:45:02.542987 C: Received prepare for C-SS                                  
I 16:45:06.983291 C: Machine C-SS finished storage at (4, 0)                    
I 16:45:06.984939 C: Machine C-SS finished processing                           
I 16:45:08.260396 MPS: Resetting machine C-SS                                   
I 16:45:09.740395 C: Received prepare for C-SS                                  
I 16:45:09.740602 C: C-SS is IDLE, processing prepare                           
I 16:45:09.741353 C: Machine C-SS broken: Prepare received for C-SS with STORE operation for occupied shelf slot (4, 0), recv 1598539502500
I 16:45:09.741829 MPS: Resetting machine C-SS   
```

And here is the clips log when the old protobuf message arrives note how it is processed directly after it arrives (the `production-machine-prepare` rule causes the printout with the timestamp in the above refbox log):
```
D 16:45:09.703800 C: ==> f-2048  (protobuf-msg (type "llsf_msgs.PrepareMachine") (comp-id 2000) (msg-type 101) (rcvd-via BROADCAST) (rcvd-from "127.0.0.1" 4441) (rcvd-at 1598539509 703661) (client-type PEER) (client-id 1953722218) (ptr <Pointer-5-0x7f2c580016b0>))
D 16:45:09.738501 C: FIRE    2 ws-update-gamestate: f-2050                      
D 16:45:09.740317 C: FIRE    5 production-machine-prepare: f-2050,f-2048,f-213  
D 16:45:09.740355 C: <== f-2048  (protobuf-msg (type "llsf_msgs.PrepareMachine") (comp-id 2000) (msg-type 101) (rcvd-via BROADCAST) (rcvd-from "127.0.0.1" 4441) (rcvd-at 1598539509 703661) (client-type PEER) (client-id 1953722218) (ptr <Pointer-5-0x7f2c580016b0>))
D 16:45:09.740857 C: <== f-1956  (machine (name C-SS) (team CYAN) (mtype SS) (actual-lights GREEN-ON) (desired-lights GREEN-ON) (productions 0) (state IDLE) (prev-state IDLE) (task nil) (mps-busy FALSE) (mps-ready FALSE) (proc-time 0) (proc-start 33.280192) (down-period -1.0 -1.0) (broken-since 0.0) (broken-reason "") (pose 0.0 0.0 0.0) (pose-time 0 0) (zone C_Z68) (rotation 0) (prep-blink-start 14.559444) (idle-since 33.280192) (wait-for-product-since 11.519568) (mps-base-counter 0) (bases-added 0) (bases-used 0) (bs-side INPUT) (bs-color BASE_RED) (ds-gate 0) (ds-last-gate 0) (ds-order 0) (ss-operation STORE) (ss-shelf-slot 4 0) (rs-ring-color RING_BLUE) (rs-ring-colors RING_GREEN RING_BLUE) (cs-operation RETRIEVE_CAP) (cs-retrieved FALSE))
D 16:45:09.740918 C: ==> f-2053  (machine (name C-SS) (team CYAN) (mtype SS) (actual-lights GREEN-ON) (desired-lights GREEN-ON) (productions 0) (state BROKEN) (prev-state IDLE) (task nil) (mps-busy FALSE) (mps-ready FALSE) (proc-time 0) (proc-start 33.280192) (down-period -1.0 -1.0) (broken-since 0.0) (broken-reason "Prepare received for C-SS with STORE operation for occupied shelf slot (4, 0), recv 1598539502500") (pose 0.0 0.0 0.0) (pose-time 0 0) (zone C_Z68) (rotation 0) (prep-blink-start 14.559444) (idle-since 33.280192) (wait-for-product-since 11.519568) (mps-base-counter 0) (bases-added 0) (bases-used 0) (bs-side INPUT) (bs-color BASE_RED) (ds-gate 0) (ds-last-gate 0) (ds-order 0) (ss-operation STORE) (ss-shelf-slot 4 0) (rs-ring-color RING_BLUE) (rs-ring-colors RING_GREEN RING_BLUE) (cs-operation RETRIEVE_CAP) (cs-retrieved FALSE))
```

**Comment by morxa** [comment]
I think this is the same issue as #39.

**Comment by TarikViehmann** [comment]
resolved by #91 (or rather the symptom is fixed by attaching timestamps to messages and discarding old messages, the actual issue is still unknown but should not matter anymore)
